packages/core-node/src/create-application-metadata-provider.ts

```diff
@@ -0,0 +1,69 @@
+import { Communication, reportError } from '@wixc3/engine-core';
+import { createDisposables } from '@wixc3/patterns';
+import { MetadataCollectionAPI, metadataApiToken } from './types';
+import { METADATA_PROVIDER_ENV_ID, ENGINE_ROOT_ENVIRONMENT_ID } from './constants';
+import memoizeOne from 'memoize-one';
+
+/**
+ * creates a new instance of metadata provider that can get application metadata using `MetadataCollectionAPI` api
+ * that should be registered for `ROOT_ENGINE_ENV_ID` environment
+ * @param com The communication instance that have registered hosts for `ROOT_ENGINE_ENV_ID` and `METADATA_PROVIDER_ENV_ID`
+ */
+export function createMetadataProvider(com: Communication) {
+    const metadataProviderCom = createMetadataProviderCommunication(com);
```

Comment:
can you use `createMetadataProviderCommunication` inside `getMetadataFromCommunication` si we do not need to memoize twice?

Reply:
moved all logic to single function